airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/job_factory/OAuthConfigSupplier.java

```diff
@@ -35,25 +41,37 @@ public OAuthConfigSupplier(ConfigRepository configRepository, boolean maskSecret
   public JsonNode injectSourceOAuthParameters(UUID sourceDefinitionId, UUID workspaceId, JsonNode sourceConnectorConfig)
       throws IOException {
     try {
+      final ImmutableMap<String, Object> metadata = generateSourceMetadata(sourceDefinitionId);
       // TODO there will be cases where we shouldn't write oauth params. See
       // https://github.com/airbytehq/airbyte/issues/5989
       MoreOAuthParameters.getSourceOAuthParameter(configRepository.listSourceOAuthParam().stream(), workspaceId, sourceDefinitionId)
           .ifPresent(
-              sourceOAuthParameter -> injectJsonNode((ObjectNode) sourceConnectorConfig, (ObjectNode) sourceOAuthParameter.getConfiguration()));
+              sourceOAuthParameter -> {
+                injectJsonNode((ObjectNode) sourceConnectorConfig, (ObjectNode) sourceOAuthParameter.getConfiguration());
+                if (!maskSecrets) {
+                  TrackingClientSingleton.get().track(workspaceId, "OAuth Injection - Backend", metadata);
```

Comment:
is this the common pattern we follow when injecting the tracking client? shouldn't it be injected through the constructor for better mocking?

Reply:
👍 . otherwise tests are going to report to segment.